### PR TITLE
Implement a hold-to-confirm screen when exiting game using escape key

### LINKED ISSUE #1940: Quit confirmation at menu screen
Right now, pressing Esc at the menu screen causes lazer to quit immediately instead of showing a quit confirmation overlay.

Are there designs from @arflyte ? I assume it should look similar to the pause and fail overlays?

Pause overlay:
maxresdefault

### PR DESCRIPTION
**Pull request by peppy**
Resolves #1940.

As this is an alternative implementation, closes #2341.

### CODE REVIEW

File: osu.Game/Overlays/HoldToConfirmOverlay.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    /// An overlay which will display a black screen that dims over a period before confirming an exit action.
+    /// Action is BYO (derived class will need to call <see cref="BeginConfirm"/> and <see cref="AbortConfirm"/> from a user event).
+    /// </summary>
+    public abstract class HoldToConfirmOverlay : Container
+    {
+        public Action Action;
+
+        private Box overlay;
+
+        private const int activate_delay = 400;
```
Comment (huoyaoyuan) [inline osu.Game/Overlays/HoldToConfirmOverlay.cs]: Should this be configurable via virtual?

Comment (peppy) [reply osu.Game/Overlays/HoldToConfirmOverlay.cs]: not until we need it. keeping things standardised is best if we can.

File: osu.Game/Overlays/HoldToConfirmOverlay.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                overlay = new Box
+                {
+                    Alpha = 0,
+                    Colour = Color4.Black,
+                    RelativeSizeAxes = Axes.Both,
+                }
+            };
+        }
+
+        protected void BeginConfirm() => overlay.FadeIn(activate_delay, Easing.Out);
```
Comment (smoogipoo) [inline osu.Game/Overlays/HoldToConfirmOverlay.cs]: `.FadeIn().Finally(() => Action?.Invoke())`

Wouldn't need the `fired` bool then.

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game/Overlays/HoldToConfirmOverlay.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Overlays/HoldToConfirmOverlay.cs]: sounds good

File: osu.Game.Tests/Visual/TestCaseHoldToConfirmOverlay.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        public TestCaseHoldToConfirmOverlay()
+        {
+            bool fired = false;
+
+            var abortText = new OsuSpriteText
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Text = "Aborted!",
```
Comment (smoogipoo) [inline osu.Game.Tests/Visual/TestCaseHoldToConfirmOverlay.cs]: Why is this "Aborted!" and not "Fired!"

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**